app/routes/languages.rb

```diff
@@ -5,21 +5,18 @@ module Routes
     class Languages < Core
       get '/languages' do
         tracks = X::Track.all
+        tracks.delete_if { |t| t.repository == 'https://github.com/exercism/xSLUG' }
         active, inactive = tracks.partition { |t| t.active? }
         inactive.sort! { |a, b| b.problems.count <=> a.problems.count }
         erb :"languages/index", locals: { active: active, inactive: inactive }
       end
 
       get '/languages/:track_id' do |track_id|
         track = X::Track.find(track_id)
-        if track.implemented?
-          erb :"languages/language", locals: {
-            track: track,
-            docs: X::Docs::Launch.new(track.repository),
-          }
-        else
-          erb :"languages/not_implemented", locals: { track: track }
```

Comment:
If someone visits http://exercism.io/languages/shakespeare they see an error.  On line 29 in x/track.rb the initializer throws a no method error.


Maybe validate the response here: https://github.com/exercism/exercism.io/blob/master/x/track.rb#L12 before trying to call new

X::Xapi.get('tracks', 'shakespeare') returns [404, "{\"error\":\"No track 'shakespeare'\"}"]